hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/BaseTokenHandler.java

```diff
@@ -423,22 +424,47 @@ public static boolean isExpiryOnlyUpdateOp(@NonNull final TokenUpdateTransaction
     }
 
     /**
-     * Returns true if the given token update op is a metadata-only update op.
-     * This is needed for validating whether a token update op has admin key present on the token,
-     * to update any other fields other than metadata.
-     * For updating metadata we need signature from either admin key or metadata key
-     * @param op the token update op to check
-     * @return true if the given token update op is an metadata-only update op
+     * Check if TokenUpdateOp wants to remove some of the token keys
+     * low priority keys are -> wipeKey, kycKey, supplyKey, freezeKey, feeScheduleKey, pauseKey, metadataKey
      */
-    public static boolean isMetadataOnlyUpdateOp(@NonNull final TokenUpdateTransactionBody op) {
-        if (!op.hasMetadata()) {
-            return false;
-        }
-        final var defaultWithMetadata = TokenUpdateTransactionBody.newBuilder()
-                .metadata(op.metadata())
-                .token(op.token())
-                .build();
-        return op.equals(defaultWithMetadata);
+    public static boolean containsKeyRemoval(@NonNull final TokenUpdateTransactionBody op) {
+        return op.hasAdminKey() && isKeyRemoval(op.adminKey())
```

Comment:
If we make `isKeyRemoval(Key)` return false when it receives `null`, we can remove all these `op.has<X>Key()` checks and this becomes more readable.